### PR TITLE
headlamp-plugin: Make update check asynchronous

### PR DESCRIPTION
**Pull request by skoeva**
This change makes the update check for headlamp-plugin asynchronous to avoid slowing down and blocking the user on start.

Fixes: #1899

### CODE REVIEW

File: plugins/headlamp-plugin/bin/headlamp-plugin.js [hunk 1]
```diff
@@ -269,24 +270,46 @@ function start() {
...
-    if ('@kinvolk/headlamp-plugin' in outdated) {
-      const url = `https://github.com/kinvolk/headlamp/releases`;
-      console.warn(
-        '    @kinvolk/headlamp-plugin is out of date. Run the following command to upgrade \n' +
-          `    See release notes here: ${url}` +
-          '    npx @kinvolk/headlamp-plugin upgrade'
+  async function informIfOutdated() {
+    const lastCheckTime = process.env.LAST_UPDATE_CHECK
+      ? parseInt(process.env.LAST_UPDATE_CHECK, 10)
+      : null;
```
Comment (joaquimrocha) [inline plugins/headlamp-plugin/bin/headlamp-plugin.js:278]: This logic means lastCheckTime can be set as null, and then on line 282 it's used in arythmetic.

File: plugins/headlamp-plugin/bin/headlamp-plugin.js [hunk 2]
```diff
@@ -992,9 +1053,20 @@ yargs(process.argv.slice(2))
...
+    yargs => {
+      yargs.option('force-check', {
+        describe: 'Force the update check regardless of the last check time.',
+        type: 'boolean',
+        default: false,
+      });
+    },
+    async argv => {
+      process.exitCode = await start(argv.forceCheck);
+    }
```
Comment (joaquimrocha) [inline plugins/headlamp-plugin/bin/headlamp-plugin.js:1068]: Will this command still hang while it's started?

File: plugins/headlamp-plugin/bin/headlamp-plugin.js [hunk 3]
```diff
@@ -586,6 +609,44 @@ function getNpmOutdated() {
   return JSON.parse(result);
 }
 
+/**
+ * Use fetch to get the latest package version from the npm registry.
+ *
+ * @returns {Promise<Object>} - promise that resolves to an object containing version information
+ * if the package is outdated, or an empty object.
+ */
+async function getLatestPkgVersion() {
```
Comment (joaquimrocha) [inline plugins/headlamp-plugin/bin/headlamp-plugin.js:618]: You introduced a new way to check for the latest version. But the previous function is still used for the upgrade. Should we unify this?

**Review by joaquimrocha — CHANGES_REQUESTED** [review]
Besides the comments about the logic. I think this PR is doing 2 things: replacing the function that checks for the latest pkg version + limiting the checks.

**Review by illume — CHANGES_REQUESTED** [review]
From this comment in the issue: https://github.com/headlamp-k8s/headlamp/issues/1899#issuecomment-2099965074

> It could be made async, so it doesn't block the user getting work done.
>
> If the check is done async, then it can show the message some moments later. This avoids having state, and has the benefit that it isn't slow once a day. The user is never slowed down in their work.

Currently doing `npm start` with this branch will do the check in an async fashion never block the user. So I think all of the last check stuff is not necessary.

The main goal of not blocking the user is achieved with the code:

```ts
  informIfOutdated().catch(error => {
    console.error('Failed to check for updates:', error);
  });
```

So please keep that `informIfOutdated()` call, and remove all the `lastCheckTime` code and the `force-check` code.

**Comment by skoeva** [comment]
@illume I see, thanks for the fix! I think this makes sense since I noticed issues with performing the `getNpmOutdated()` code async, I can remove the debug lines and squash this now

**Review by joaquimrocha — APPROVED** [review]

**Pull request merged**